script.js

```diff
@@ -308,12 +287,65 @@ function createOptions([events, eventsNextYear]) {
   });
   select.addEventListener("change", (event) => {
     const eventDate = DateTime.fromISO(`${event.target.value}`);
+    addTimeToQuery(eventDate.toMillis());
     const eventDifference = getTimeDifference(DateTime.local(), eventDate);
-    document.getElementById("timer").remove();
-    createTimer();
-    setTimeValues(eventDifference);
-    timelines.forEach((timeline) => timeline.kill());
-    createTimeline(timeUnits.seconds, 0);
+    setupTimer(eventDifference);
   });
   document.body.appendChild(select);
 }
+function setupTimer(eventDifference) {
+  document.getElementById("timer")?.remove();
+  createTimer();
+  setTimeValues(eventDifference);
+  if (timelines.length) {
+    timelines.forEach((timeline) => timeline.kill());
+  }
+  createTimeline(timeUnits.seconds, 0);
+  document.addEventListener("visibilitychange", onVisibilityChange);
+}
+
+/**
+ * Retrieves the event time from the query parameters in the URL. If not found
+ * or not able to parse, returns an object with all properties set to 0. The
+ * time is expected to be in milliseconds.
+ * @returns {Object} An object representing the event time, with properties for
+ *  years, months, days, hours, minutes, and seconds.
+ */
+function getTimeFromQuery() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const eventDate = urlParams.get("time");
+  try {
+    if (eventDate) {
+      return getTimeDifference(
+        DateTime.local(),
+        DateTime.fromMillis(parseInt(eventDate)),
+      );
+    }
+  } catch (error) {
+    console.error("Error parsing event time:", error);
+  }
+  return null;
+}
+
+/**
+ * Adds the specified time to the query string of the current URL.
+ * @param {string} time - The time to be added to the query string, 
+ *  in milliseconds.
+ */
+function addTimeToQuery(time) {
```

Comment:
@Danil4UA When the user changes the date, the URL should be updated using this function. You should convert it to Unix time by using Luxon's [fromISO](https://github.com/moment/luxon/blob/master/docs/parsing.md#parsing-technical-formats), see example [here](https://github.com/fcFn/DI_Hackathon_1/pull/19/files#diff-ed3ee7e0beea2498ff3b8ca85973d122fc6fa3d585d62b5807ec034d0cf076b3R289).

Reply:
I've done that while integrating your changes :)